spec/models/manageiq/providers/kubevirt/inventory/parser_spec.rb

```diff
@@ -0,0 +1,110 @@
+  describe '#process_templates' do
+    it 'parses a template' do
+      disk_collection = double("disk_collection")
+      disk = FactoryGirl.create(:disk)
+      allow(disk_collection).to receive(:find_or_build_by).and_return(disk)
+
+      hw_collection = double("hw_collection")
+      hardware = FactoryGirl.create(:hardware)
+      allow(hw_collection).to receive(:find_or_build).and_return(hardware, :disks => [disk])
+
+      os_collection = double("os_collection")
+      os = FactoryGirl.create(:operating_system)
+      allow(os_collection).to receive(:find_or_build).and_return(os)
+
+      template_collection = double("template_collection")
+      temp = FactoryGirl.create(:template_kubevirt, :hardware => hardware, :operating_system => os)
+      allow(template_collection).to receive(:find_or_build).and_return(temp)
+
+      template_hash = {
+        :apiVersion => "kubevirt.io/v1alpha1",
+        :kind       => "Template",
+        :metadata   => {:annotations       =>
+                                              {:description => "OCP kubevirt linux, template",
+                                               :tags        => "kubevirt,ocp,template,linux"},
+                        :clusterName       => "",
+                        :creationTimestamp => "2018-01-24T10:15:25Z",
+                        :labels            =>
+                                              {:"miq.github.io/kubevirt-is-vm-template" => true,
+                                               :os                                      => "rhel-7"},
+                        :name              => "test",
+                        :namespace         => "default",
+                        :resourceVersion   => "262502",
+                        :selfLink          => "/apis/kubevirt.io/v1alpha1/namespaces/default/templates/test",
+                        :uid               => "7e6fb1ac-00ef-11e8-8840-525400b2cba8"},
+        :spec       => {:template =>
+                                     {:spec =>
+                                               {:domain  =>
+                                                            {:cpu       => {:cores => 4},
+                                                             :devices   => {:disks => [{:disk       => {:dev => "vda"},
+                                                                                        :name       => "disk0",
+                                                                                        :volumeName => "disk0-pvc"}]},
+                                                             :machine   => {:type     => "q35"},
+                                                             :resources => {:requests => {:memory => "4096Mi"}}},
+                                                :volumes => [{:name                  => "disk0-pvc",
+                                                              :persistentVolumeClaim => {:claimName => "linux-vm-pvc-test"}}]}}}
+      }
```

Comment:
The indentation used in this hash ^ makes it hard to read. Wouldn't it be better to just use two spaces for indentation?

```ruby
template_hash = {
  :metadata => {
    :annotations => {
      ...
    }
  }
}
```

Did you consider creating a `spec/fixtures/....json` file and reading it like we do for the `one_node.json` file used in the tests for the full refresh?

Reply:
Good idea I will use `fixtures`. I would be more readable.